# Case Study on Testing of Hypothesis

In [1]:
import numpy as np
import pandas as pd
from scipy.stats import ttest_rel
from scipy import stats

## A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.

In [2]:
sales_data = pd.read_csv('Sales_add.csv')

In [3]:
sales_data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [ ]:
# Rename the Columns 

In [4]:
sales_data.columns=['Month', 'Region', 'Manager', 'Sales_before_digital', 'Sales_After_digital']
sales_data

Month      Region      Manager  Sales_before_digital  \
0    Month-1  Region - A  Manager - A                132921   
1    Month-2  Region - A  Manager - C                149559   
2    Month-3  Region - B  Manager - A                146278   
3    Month-4  Region - B  Manager - B                152167   
4    Month-5  Region - C  Manager - B                159525   
5    Month-6  Region - A  Manager - B                137163   
6    Month-7  Region - C  Manager - C                130625   
7    Month-8  Region - A  Manager - A                131140   
8    Month-9  Region - B  Manager - C                171259   
9   Month-10  Region - C  Manager - B                141956   
10  Month-11  Region - B  Manager - C                159339   
11  Month-12  Region - A  Manager - A                178939   
12  Month-13  Region - B  Manager - A                145062   
13  Month-14  Region - A  Manager - A                151514   
14  Month-15  Region - C  Manager - A                147463   
15  Month-16  Region - A  Manager - B                177195   
16  Month-17  Region - A  Manager - C                140862   
17  Month-18  Region - C  Manager - B                167996   
18  Month-19  Region - B  Manager - A                132135   
19  Month-20  Region - A  Manager - B                152493   
20  Month-21  Region - B  Manager - A                147425   
21  Month-22  Region - A  Manager - C                130263   

    Sales_After_digital  
0                270390  
1                223334  
2                244243  
3                231808  
4                258402  
5                256948  
6                222106  
7                230637  
8                226261  
9                193735  
10               203175  
11               276279  
12               205180  
13               253131  
14               229336  
15               187305  
16               234908  
17               191517  
18               227040  
19               212579  
20               263388  
21               243020

## ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing

   Let u1 = mean of Sales_After_digital and u2 = mean of Sales_before_digital

- H0 (Null hypothesis) : (u1<=u2) There is no significant increase in sales after stepping into digital marketing. 
- Ha (Alternative hypothesis) : (u1>u2)  There is a significant increase in sales after stepping into digital marketing 

 Here we need to perform two sample paired t-test

In [8]:
ttest_rel(sales_data['Sales_After_digital'],sales_data['Sales_before_digital'], alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

 Here we could see that the p-value = 3.168e-11 is significantly smaller than 0.05.

 - Hence we reject the Null Hypothesis H0.

Therefore, There is a significant increase in sales after stepping into digital marketing.

## ● The company needs to check whether there is any dependency between the features “Region” and “Manager”

Inorder to check whether there is any dependency between the features “Region” and “Manager”, we need to use Chi-Sqaure test of independence

In [11]:
sales_data

Month      Region      Manager  Sales_before_digital  \
0    Month-1  Region - A  Manager - A                132921   
1    Month-2  Region - A  Manager - C                149559   
2    Month-3  Region - B  Manager - A                146278   
3    Month-4  Region - B  Manager - B                152167   
4    Month-5  Region - C  Manager - B                159525   
5    Month-6  Region - A  Manager - B                137163   
6    Month-7  Region - C  Manager - C                130625   
7    Month-8  Region - A  Manager - A                131140   
8    Month-9  Region - B  Manager - C                171259   
9   Month-10  Region - C  Manager - B                141956   
10  Month-11  Region - B  Manager - C                159339   
11  Month-12  Region - A  Manager - A                178939   
12  Month-13  Region - B  Manager - A                145062   
13  Month-14  Region - A  Manager - A                151514   
14  Month-15  Region - C  Manager - A                147463   
15  Month-16  Region - A  Manager - B                177195   
16  Month-17  Region - A  Manager - C                140862   
17  Month-18  Region - C  Manager - B                167996   
18  Month-19  Region - B  Manager - A                132135   
19  Month-20  Region - A  Manager - B                152493   
20  Month-21  Region - B  Manager - A                147425   
21  Month-22  Region - A  Manager - C                130263   

    Sales_After_digital  
0                270390  
1                223334  
2                244243  
3                231808  
4                258402  
5                256948  
6                222106  
7                230637  
8                226261  
9                193735  
10               203175  
11               276279  
12               205180  
13               253131  
14               229336  
15               187305  
16               234908  
17               191517  
18               227040  
19               212579  
20               263388  
21               243020

In [13]:
rm=pd.crosstab(sales_data.Region,sales_data.Manager)
rm

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

Now the could find the crosstab frequencies of the two groups.

Let,

- H0 (Null Hypothesis): There is no dependency between Region and Manager
- Ha (Alternative Hypothesis): There is a dependency between Region and Manager

For Chi-square testing use the below syntax

In [14]:
stats.chi2_contingency(rm)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

The first value (3.050) is the Chi-square value, followed by the p-value (0.54939), then comes the degrees of freedom (4), and lastly it outputs the expected frequencies as an array.

Here we could see that the p-value = 0.549 is significantly larger than 0.05.

- Hence we fail to reject the Null Hypothesis H0.

Therefore, There is no dependency between Region and Manager.